pimd/pim_bsm.c

```diff
@@ -212,36 +255,132 @@ static inline void pim_bs_timer_restart(struct bsm_scope *scope, int bs_timeout)
 	pim_bs_timer_start(scope, bs_timeout);
 }
 
+static void bsm_unicast_sock_read(struct event *t)
+{
+	struct bsm_scope *scope = EVENT_ARG(t);
+	struct sockaddr_storage from;
+	struct sockaddr_storage to;
+	socklen_t fromlen = sizeof(from);
+	socklen_t tolen = sizeof(to);
+	ifindex_t ifindex = 0;
+	struct interface *ifp;
+	uint8_t buf[PIM_PIM_BUFSIZE_READ];
+	int len, i;
+
+	event_add_read(router->master, bsm_unicast_sock_read, scope,
+		       scope->unicast_sock, &scope->unicast_read);
+
+	for (i = 0; i < router->packet_process; i++) {
```

Comment:
Line 273 to Line 320 is same code as in function pim_sock_read, wont it be better if we can make a common function and reuse the code.

Reply:
done. Added as a separate commit for easy review.